### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by rabayarabu**
### In this project
- I created a class `Nameable`.
- Implement a method called `correct_name `that will raise a `NotImplementedError`.
- Implement the `correct_name `method that returns the result of the `correct_name `method of the `@nameable`.
- Create a `CapitalizeDecorator `class that inherits from the base Decorator class.
- Implement a method `correct_name `that capitalizes the output of `@nameable.correct_name`.
- Create a `TrimmerDecorator  `class that inherits from the base Decorator class.
- Implement a method `correct_name `that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.
- Update Person class.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -15,7 +21,7 @@ def can_use_services?
 
   private
 
-  def of_age?
-    @age >= 18
+  def correct_name
+    @name
   end
 end
```
Comment (V-Blaze) [inline person.rb]: - You have done great so far but there are few issues you need to fix before moving to the next milestone. :smile: :wave: 
- [ ] The method `correct_name` is not a private method and therefore it should be moved above the `private` keyword. :+1:

File: person.rb [hunk 2]
```diff
@@ -1,8 +1,14 @@
-class Person
+require_relative 'base_decorator'
+require_relative 'capitalize_decorator'
+require_relative 'nameable'
+require_relative 'trimmer_decorator'
```
Comment (V-Blaze) [inline person.rb]: - [ ] According to the project requirement, you are meant to run this test to see your decorator in action.

```sh
person = Person.new(22, 'maximilianus')
person.correct_name
capitalized_person = CapitalizeDecorator.new(person)
puts capitalized_person.correct_name
capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)
puts capitalized_trimmed_person.correct_name
```
-  [ ] When I ran these, I got an error in the console, I'll suggest you go through your classes and make sure everything is implemented correctly, and then run the code to make sure the result you get is whats expected. :smile: :small_airplane: :wave: 

- - Screenshot for reference:
Screenshot from 2023-08-29 10-37-54

File: trimmer_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require_relative 'base_decorator'
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    super[0..9]
+  end
+end
```
Comment (V-Blaze) [inline trimmer_decorator.rb:7]: - [ ] This was supposed to be pointed out by the previous reviewers, you are meant to remove this Microverse logo and the text as they are placeholders that came with the template.You can add your logo if you have any or leave out the logo entirely. :small_airplane: :wave: 

- - Screenshot:
Screenshot from 2023-08-29 10-44-25

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @rabayarabu 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- All required classes created ✔
-  Some class methods implemented correctly 👋 
-  Correct git flow✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: base_decorator.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+require_relative 'nameable'
+
+class BaseDecorator < Nameable
+  attr_accessor :nameable
+
+  def initilize(nameable)
```
Comment (julie-ify) [inline base_decorator.rb]: - [ ] You have a typo error here. This should be `initialize`. You are missing an "a".

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi @rabayarabu,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @rabayarabu,

## APPROVED 🟢

## Highlights
✔️ Class decorator

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**